templates/gitlab.rb.j2

```diff
@@ -33,11 +28,26 @@ gitlab_rails['ldap_base'] = '{{ gitlab_ldap_base }}'
 
 # GitLab Nginx
 ## See https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/doc/settings/nginx.md
+nginx['listen_https'] = {{ 'true' if gitlab_nginx_enabled == true else 'false' }}
+{% if gitlab_nginx_enabled == true %}
 {% if gitlab_nginx_listen_port is defined %}
 nginx['listen_port'] = "{{ gitlab_nginx_listen_port }}"
 {% endif %}
 {% if gitlab_nginx_listen_https is defined %}
-nginx['listen_https'] = "{{ gitlab_nginx_listen_https }}"
+nginx['listen_https'] = {{ 'true' if gitlab_nginx_listen_https == true else 'false' }}
+{% endif %}
+{% if gitlab_custom_server_config is defined %}
+nginx['custom_gitlab_server_config'] = '{{gitlab_custom_server_config}}'
+{% endif %}
+{% if gitlab_real_ip_trusted_addresses is defined %}
```

Comment:
I like that approach! Will apply that pattern to my other roles as well - thanks